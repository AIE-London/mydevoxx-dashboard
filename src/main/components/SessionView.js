import React, {Component} from 'react';
import {Row, Col} from 'react-flexbox-grid';
import styled from 'styled-components';
import Report from './Report';

// Styles for the different elements of the page
const DayText = styled.h2`
   font-size: xx-large;
   text-indent: 50px;
   color: orange;
   margin: 5px 0;
`;

const WhiteText = styled.h2`
  font-size: large;
  text-indent: 50px;
  font-weight: 100;
  color: white;
`;

const Window = styled.div`
  text-align: left;
`;

class SessionView extends Component {
    render() {
        return (
            <Window>

                <DayText>Day {this.props.dayNo}</DayText>

                <WhiteText>{this.props.sTime} - {this.props.room}</WhiteText>

                <Row center="xs">
                    <Col xs={10}>
                        <Report>{this.props.report}</Report>
                    </Col>
                </Row>
            </Window>
        );
    }
}

export default SessionView;
