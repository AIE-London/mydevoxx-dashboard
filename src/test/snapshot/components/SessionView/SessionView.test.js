import React from 'react';
import ReactDOM from 'react-dom';
import SessionView from '../../../../main/components/SessionView';
import renderer from 'react-test-renderer';

test('app component', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SessionView  dayNo="One" sTime="10:00" room="Mezzanine"/>,
            div);
    });

});

test('card component with title snapshot', () => {

  const tree = renderer.create(
    <SessionView dayNo="One" sTime="10:00" room="Mezzanine" />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});