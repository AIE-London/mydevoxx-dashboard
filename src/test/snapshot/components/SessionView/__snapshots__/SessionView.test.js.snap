exports[`test card component with title snapshot 1`] = `
<div
  className="dZUSMd">
  <h2
    className="dzXVgj">
    Day 
    One
  </h2>
  <h2
    className="iHLkzq">
    10:00
     - 
    Mezzanine
  </h2>
  <div
    className="row center-xs">
    <div
      className="col-xs-10">
      <div
        className="bvnlUe row start-xs">
        <div
          className="col-xs-6 fteZLe">
          <div
            className="row start-xs">
            <h2
              className="Axdp"
              id="title">
              Welcome to Devoxx 2017
            </h2>
          </div>
          <div
            className="row end-xs">
            <div
              className="row"
              left="xs">
              <div
                className="kmDnAn" />
              <div
                className="kmDnAn" />
              <div
                className="kmDnAn" />
              <div
                className="kmDnAn" />
              <div
                className="cWcebC" />
            </div>
          </div>
          <p
            className="jLJUnn">
            Join the organisers of Devoxx UK and great keynote speakers for inspring stories in 20 minute segments.
          </p>
          <div
            className="row">
            <h3
              className="jJURiy">
              Top Tracks
            </h3>
          </div>
          <div
            className="row">
            <ul
              className="izGhAk">
              <li
                className="cLQKgI">
                Java
              </li>
              <li
                className="cLQKgI">
                Devoxx
              </li>
              <li
                className="cLQKgI">
                Spring
              </li>
            </ul>
          </div>
        </div>
        <div
          className="col-xs-6 eZvXOT">
          <div
            className="row start-xs">
            <h3
              className="jJURiy"
              id="notesHeader">
              My Notes
            </h3>
          </div>
          <div
            className="row start-xs">
            <p
              className="bKHqKP">
              Lorem ipsum dolor sit amet, everti quaestio mel ea. Ex eos volutpat qualisque. Sale tantas cotidieque quo ut, ad nostro consectetuer nec. Feugiat qualisque quo an. Labores officiis te nam.
            </p>
          </div>
          <div
            className="row start-xs">
            <h3
              className="jJURiy"
              id="reviewHeader">
              My Reviews
            </h3>
          </div>
          <div
            className="fEZGiH row start-xs">
            <div
              className="row start-xs">
              <div
                className="">
                <img
                  className="imhmHM"
                  src="test-image.jpeg" />
              </div>
              <div
                className="col-xs">
                <h3
                  className="jPwRym">
                  Test User
                </h3>
                <p
                  className="ZxTsc">
                  Great session, thanks for organising. Looking forward to the next one!
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-10">
      <div
        className="lluTUT row">
        <div
          className="kblate">
          <div
            className="row">
            <div
              className="">
              <img
                className="jmFtVx"
                src="defaultIcon.png" />
            </div>
            <div
              className="col-xs">
              <div>
                <h3>
                  Test Speaker
                </h3>
                <p>
                  Capgemini
                </p>
              </div>
            </div>
          </div>
          <div
            className="row">
            <div
              className="">
              <div>
                Personal Blog:
              </div>
              <div
                className="fYLUow">
                personalblog.com
              </div>
            </div>
          </div>
          <div
            className="row">
            <div
              className="">
              <div>
                Other Talks:
              </div>
              <ul>
                <div
                  className="fYLUow">
                  <li>
                    Intro to Devoxx (Room 1 - 11:45)
                  </li>
                  <li>
                    Intro to Devoxx 2 (Room 2 - 13:45)
                  </li>
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
